packages/material-ui-styles/test/index.spec.tsx

```diff
@@ -99,6 +99,40 @@ function testGetThemeProps(theme: Theme, props: AppBarProps): void {
     // but this would pass anyway
     const alsoClasses = useUnsafeProps(undefined); // runtime: Can't read property color of undefined
   };
+
+  // default theme
+  interface CustomTheme {
+    attribute: number;
+    [key: string]: any;
+  }
+
+  const validCustomTheme = { attribute: 8, otherStuff: true };
+  const invalidCustomTheme = { otherStuff: true };
+
+  const style = (theme: CustomTheme) =>
+    createStyles({
```

Comment:
That was a tricky bug. Since we also handle props now it's impossible to tell if a given function depends on the theme or props. Some bug was triggered but the error message wasn't helpful because the types were completely lost with that pattern. The theme should not be referenced inside `createStyles`. I guess we should add this to the docs.